kong/vault_service_test.go

```diff
@@ -0,0 +1,207 @@
+package kong
+
+import (
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVaultsService(t *testing.T) {
+	RunWhenEnterprise(t, ">=3.0.0", RequiredFeatures{})
+	assert := assert.New(t)
+	require := require.New(t)
+
+	client, err := NewTestClient(nil, nil)
+	assert.NoError(err)
+	assert.NotNil(client)
+
+	vault := &Vault{
+		Name:        String("env"),
+		Prefix:      String("my-env-vault"),
+		Description: String("ENV vault for secrets"),
+		Config: Configuration{
+			"prefix": "SECURE_",
+		},
+	}
+
+	createdVault, err := client.Vaults.Create(defaultCtx, vault)
+	assert.NoError(err)
+	require.NotNil(createdVault)
+
+	vault, err = client.Vaults.Get(defaultCtx, createdVault.ID)
+	assert.NoError(err)
+	assert.NotNil(vault)
+
+	vault.Prefix = String("my-new-env-vault")
+	vault.Description = String("new ENV vault for secrets")
+
+	vault, err = client.Vaults.Update(defaultCtx, vault)
+	assert.NoError(err)
+	assert.NotNil(vault)
+	assert.Equal("env", *vault.Name)
+	assert.Equal("new ENV vault for secrets", *vault.Description)
+	assert.Equal("my-new-env-vault", *vault.Prefix)
+
+	err = client.Vaults.Delete(defaultCtx, vault.ID)
+	assert.NoError(err)
+
+	// ID can be specified
+	id := uuid.NewString()
+	vault = &Vault{
+		Name:        String("aws"),
+		Prefix:      String("my-aws-vault"),
+		ID:          String(id),
+		Description: String("aws vault for secrets"),
+		Config: Configuration{
+			"region": "us-east-2",
+		},
+	}
+
+	createdVault, err = client.Vaults.Create(defaultCtx, vault)
+	assert.NoError(err)
+	assert.NotNil(createdVault)
+	assert.Equal(id, *createdVault.ID)
+	assert.Equal("aws", *createdVault.Name)
+	assert.Equal("aws vault for secrets", *createdVault.Description)
+	assert.Equal(Configuration{"region": "us-east-2"}, createdVault.Config)
+
+	err = client.Vaults.Delete(defaultCtx, createdVault.ID)
+	assert.NoError(err)
+
+	_, err = client.Vaults.Create(defaultCtx, nil)
+	assert.NotNil(err)
+
+	_, err = client.Vaults.Update(defaultCtx, nil)
+	assert.NotNil(err)
```

Comment:
Do you mind adding error messages to those assertions? Something like `an update called with the same object as create shouldn't fail` (or similar that would fit here).

Without those I feel those are a bit without context.

Alternatively what you could do is to put those into subtests with proper names.

Reply:
Sure, I agree these asserts checking the calls error out are a bit obscure without hints.